addons/account/views/account_portal_templates.xml

```diff
@@ -113,9 +113,39 @@
                 <t t-call="portal.portal_record_sidebar">
                     <t t-set="classes" t-value="'col-lg-3 col-xl-4 d-print-none me-lg-auto'"/>
                     <t t-set="title">
+                        <t t-set="amount" t-value="invoice.amount_residual if invoice.amount_residual > 0 else invoice.amount_total"/>
+
+                        <t t-if="invoice.invoice_payment_term_id.early_discount">
+                            <t t-if="invoice.payment_state in ['paid', 'in_payment']">
+                                <t t-set="done_tx" t-value="invoice.transaction_ids.filtered(lambda tx: tx.state == 'done')[:1]"/>
+                                <t t-if="done_tx"
+                                    t-set="was_paid_with_early_discount"
+                                    t-value="invoice.invoice_payment_term_id._get_last_discount_date(invoice.date) >= done_tx.create_date.date()"/>
+                            </t>
+
+                            <t t-set="is_eligible_for_early_payment_discount"
+                                t-value="not invoice.invoice_date
+                                    or invoice._is_eligible_for_early_payment_discount(invoice.currency_id, datetime.date.today())"/>
+
+                            <t t-set="discounted_amount"
+                                t-value="invoice.invoice_payment_term_id._get_amount_due_after_discount(amount, invoice.amount_tax)"
+                                t-if="is_eligible_for_early_payment_discount or was_paid_with_early_discount"/>
+                        </t>
+
+                        <div t-if="is_eligible_for_early_payment_discount" class="alert alert-info" role="alert">
+                            An early payment discount is available for this invoice!
+                        </div>
                         <h2>
-                            <span t-if="invoice.amount_residual > 0" t-field="invoice.amount_residual"/>
```

Comment:
usually adding new things is ok in view (so the rest should be ok)

but removing this t-field could be risky for possible XPATH override in views, so if we wanted to do this in stable it would be better if it was more stable, for example in stable you could keep the original code and have a `t-if="0"` on the original h2, this would make the view ugly (and possibly hide things that customization could add in the H2 tag) but at least this should not break any XPATH.